linopy/model.py

```diff
@@ -998,6 +998,12 @@ def solve(
             Tuple containing the status and termination condition of the
             optimization process.
         """
+        # clear cached matrix properties from potentially present from previous solve commands
+        for cached_prop in ["flat_vars", "flat_cons", "sol", "dual"]:
+            # check existence of cached_prop without creating it
+            if cached_prop in self.matrices.__dict__:
+                delattr(self.matrices, cached_prop)
+
```

Comment:
@apfelix would you mind to extract this into a function in MatrixAccessor in matrices.py? something like `clear_cache` which can then be called here.

Reply:
yes, good call, will do so